### ISSUE TITLE
Cannot focus or type in dialog input when using bootstrap and tabindex="-1" for accessibility 

### ISSUE THREAD

**Issue by OlivierJaquemet** [tinymce/tinymce#5169]
**Current behavior / Bug description**
When designing an HTML modal, `role="dialog"` and `tabindex="-1"` are required for accessibility (to ensure _focusable_ elements and inputs are only the one which are part of the modal).
Problem : when TinyMCE is used in such modal, the inputs display in TinyMCE dialog cannot be focused and entering text is impossible. 

![reproduce](https://user-images.githubusercontent.com/618861/65227577-2b39ca00-dac9-11e9-9997-53c11bde4b67.gif)

The technical explanation is that TinyMCE dialogs DOM element are placed under the body, not inside the modal. And because the [`enforceFocus()`](https://github.com/twbs/bootstrap/blob/master/js/src/modal/modal.js#L295) method of bootstrap only considers elements descendant of the modal as being _focusable_, tinymce input/dialog cannot get focus.

**Steps to reproduce**
Use the following Fiddle, created by @spocke in 2017 for issues #3421, which exhibit the problem with edge version of TinyMCE : 
http://fiddle.tinymce.com/qGfaab

**What is the expected behavior?**
Typing text in those input should be possible while still maintaining the accessibility option.

Suggested solution (without dependency to bootstrap of course) : if the TinyMCE editor is inside a modal element (ie an element with `role="dialog"` and `tabindex="-1"`) it should be used by TinyMCE to place any additional HTML element created for dialogs (it should have no impact, as modal styling will already guarantee correct UI/z-index placing without having to place under body).

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**
All.

**Comment by lnewson** [comment]
The fiddle linked was created for v4, and if you read the latest [docs](https://www.tiny.cloud/docs/integrations/bootstrap/) then you'll see the selectors required for bootstrap integration have changed. So you'll see in the updated fiddle linked on the docs, that it works as expected: http://fiddle.tinymce.com/oxdaab

**Comment by lnewson** [comment]
Correction sorry, it looks like the fiddle hasn't been updated for that page. I'll log an internal docs ticket to get that fixed. In the meantime, here's a correctly updated fiddle: http://fiddle.tinymce.com/oxdaab/669

**Comment by OlivierJaquemet** [comment]
Thank you @lnewson for your answer. Perfect. 👍🏻
